pomerium/config.go

```diff
@@ -113,6 +113,9 @@ func applySetOtherOptions(_ context.Context, p *pb.Config, c *model.Config) erro
 			Values: *c.Spec.AuthorizeLogFields,
 		}
 	}
+	if c.Spec.PassIdentityHeaders != nil {
+		p.Settings.PassIdentityHeaders = proto.Bool(*c.Spec.PassIdentityHeaders)
+	}
```

Comment:
```suggestion
	p.Settings.PassIdentityHeaders = c.Spec.PassIdentityHeaders
```

as this function merges changes with protobuf, we also want to unset that flag it it was set previously

Reply:
in fact, we _used to_ merge with current settings, but we don't anymore. as a follow-up we should update the applySettings function to explicitly return a new pb.Settings rather then taking it as an argument, that is a legacy of previous behaviour where it merged changes into it.